### DISCUSSION TITLE
Stopwatch with ui.timer

### DISCUSSION THREAD

**Discussion by Kowersky** [zauberzeug/nicegui#3576] [Q&A] [answered]
### Question

Hi. I need to do stopwatch with start button. In lambda I cant add variable and idk how to do this.

```py
global licznik_kopania
ui.separator().classes('w-[250px] h-[2px]')
licznik_kopania = ui.label('Czas Kopania: ').classes('text-h6')
ui.timer(1.0, lambda: [licznik_kopania.set_text(f"Czas Kopania: {datetime.datetime.now():%X}")])
```

**Comment by falkoschindler** [comment]
@Kowersky What do you mean with "stoper"? What do you want to achieve? And how does the separator relate to your question?
By the way, it might help to post example code with English variable names and labels, so that it better speaks for itself.

**Reply by Kowersky** [reply]
In Polish stopwatch is “stopwatch” and I thought it was the same in English.

**Comment by falkoschindler** [comment, marked as answer]
I see. Here is a simply stopwatch:

```py
label = ui.label('0.00')

def start():
    global start_time
    start_time = time.time()
    timer.active = True

def stop():
    timer.active = False

timer = ui.timer(0.1, lambda: label.set_text(f'{time.time() - start_time:.2f}'), active=False)

ui.button('Start', on_click=start)
ui.button('Stop', on_click=stop)
```
